### PR TITLE
HW3

### PR DESCRIPTION
**Pull request by ArazureX**
image

### CODE REVIEW

File: dags/process_sales.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    raw_folder_path = os.path.join(raw_data_path, raw_folder_name)
+    date_str=date.strftime('%Y-%m-%d')
+    print("Starting job1:")
+    resp = requests.post(
+        url=f'http://172.22.0.8:{JOB1_PORT}/fetch_api_data',
+        json={
+            "date": date_str,
+            "raw_dir": raw_folder_path
+        }
+    )
```
Comment (mikolaz27) [inline dags/process_sales.py:32]: remove prints, use logs

File: dags/process_sales.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+def run_job1(date, **kwargs):
+    raw_data_path = kwargs.get('raw_data_path')
+    raw_folder_name = date.strftime('%Y-%m-%d')
+    raw_folder_path = os.path.join(raw_data_path, raw_folder_name)
+    date_str=date.strftime('%Y-%m-%d')
+    print("Starting job1:")
+    resp = requests.post(
+        url=f'http://172.22.0.8:{JOB1_PORT}/fetch_api_data',
+        json={
+            "date": date_str,
```
Comment (mikolaz27) [inline dags/process_sales.py:29]: use hooks

File: dags/process_sales.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    print("job1 completed!")
+  
+
+def run_job2(date, **kwargs):
+    raw_data_path = kwargs.get('raw_data_path')
+    stg_data_path = kwargs.get('stg_data_path')
+    raw_folder_name = date.strftime('%Y-%m-%d')
+    raw_folder_path = os.path.join(raw_data_path, raw_folder_name)
+    stg_folder_name = date.strftime('%Y-%m-%d')
+    stg_folder_path = os.path.join(stg_data_path, stg_folder_name)
```
Comment (mikolaz27) [inline dags/process_sales.py:43]: the same, remove prints use hooks

File: dags/process_sales.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        default_args=default_args,
+        schedule = "0 1 * * *",
+        tags=['HM_3'],
+        catchup=True
+) as dag:
+    while processed_days < days_to_process:
+        date_to_process = start_date - timedelta(days=processed_days)
+        date_to_process_str = date_to_process.strftime('%Y-%m-%d')
+        extract_data_from_api = PythonOperator(
+            task_id=f'extract_data_for_date_{date_to_process_str}_iter_{processed_days}',
```
Comment (mikolaz27) [inline dags/process_sales.py:80]: one run - one date, remove this while

File: dags/process_sales.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            python_callable=run_job1,
+            op_kwargs={'date': date_to_process, 'raw_data_path': RAW_DIR},
+            provide_context=True
+    )
+   
+        convert_to_avro = PythonOperator(
+            task_id=f'convert_data_for_date_{date_to_process_str}_iter_{processed_days}',
+            python_callable=run_job2,
+            op_kwargs={'date': date_to_process, 'raw_data_path': RAW_DIR, 'stg_data_path': STG_DIR},
```
Comment (mikolaz27) [inline dags/process_sales.py:89]: you can use httpoperator or hook

**Review by ArazureX — COMMENTED** [review]
ОК

**Pull request merged**